collector/processes_linux.go

```diff
@@ -95,15 +98,22 @@ func getAllocatedThreads() (int, map[string]int32, int, error) {
 	if err != nil {
 		return 0, nil, 0, err
 	}
+	pids := 0
 	thread := 0
 	procStates := make(map[string]int32)
 	for _, pid := range p {
 		stat, err := pid.NewStat()
+		// PIDs can vanish between getting the list and getting stats.
+		if os.IsNotExist(err) {
+			log.Debugf("file not found when retrieving stats: %q", err)
```

Comment:
Just use `%s` or `%v` for `error` type formatting (but `%q` is good to quote file names embedded in a string).

Reply:
This is an issue all over the codebase.  I filed https://github.com/prometheus/node_exporter/issues/1044 to do cleanup.

Reply:
It seems like all the other `%q` I found are legit - `%q` makes sense when you actually want to quote something, but when you have a final colon and then an entire error string following, it doesn't make sense to quote that.